src/main/java/org/jboss/pnc/api/enums/OperationStatus.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014-2020 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.api.enums;
+
+/**
+ * Status of a running/completed Operation.
+ * 
+ * The OperationStatus class is used by all the Operation sub-classes (e.g. DeliverablesAnalyzerOperation)
+ *
+ */
+public enum OperationStatus {
+
+    NEW, IN_PROGRESS, SUCCESSFUL, FAILED, CANCELLED, TIMEOUT, SYSTEM_ERROR;
```

Comment:
Do we need all this statuses, can we merge some of the status enums we already have?
This case seems pretty similar to the `BuildProgress`. It would be nice if can use one enum for both cases.

Reply:
The `BuildProgress` does not have a FAILED or SYSTEM_ERROR, so it would not be sufficient. `BuildStatus` has plenty and we could reuse it. I think we would need at least IN_PROGRESS (aka BUILDING), SUCCESSFUL (aka SUCCESS), FAILED, SYSTEM_ERROR  ,and maybe CANCELLED ?

Reply:
Now we are making this one a generic one, IMO this one should deprecate other statues. We also have `ResultStatus` which is generic, but it's only about the results, not the progress.
Does the `OperationStatus` need all of those or should it have only NEW, IN_PROGRESS, COMPLETED and use the `ResultStatus` to see how it completed?

The difference with the build operation is that the build has dependencies.

Reply:
So not sure if that is what you are proposing. But I could have a progressStatus (using enums from `JobNotificationProgress`) and a resultStatus (using enums from `ResultStatus`). This way we would get rig of thr new `OperationStatus`. wdyt?

Reply:
I looks like we cannot unify all the status enums we have now. I agree with the the proposal for progressStatus and more detailed resultStatus.

JobNotificationProgress has the same values as BuildProgress, unfortunately both names are not the best. What about copying the BuildProgress to OperationStatus and deprecate the other two?

ResultStatus: unrestricted string value, which can be converted to an enum based on the operation type?

Reply:
> I looks like we cannot unify all the status enums we have now. I agree with the the proposal for progressStatus and more detailed resultStatus.
> JobNotificationProgress has the same values as BuildProgress, unfortunately both names are not the best. What about copying the BuildProgress to OperationStatus and deprecate the other two?

+1

> ResultStatus: unrestricted string value, which can be converted to an enum based on the operation type?

Not sure about this one. What are the different/conflicting/unmergeble/... values that you would like in different enums?